File: 3rd Sem/DS/2_1.c
// Write a program to implement Binary Search.
#include <stdio.h>

int binarySearch(int arr[], int l, int r, int a) {
    while (l <= r) {
        int mid = l + (r - l) / 2; 

        if (arr[mid] == a) {
            return mid; 
        } else if (arr[mid] < a) {
            l = mid + 1; 
        } else {
            r = mid - 1; 
        }
    }

    return -1; 
}

int main() {
    int size, a;
    
    printf("Enter the size of the sorted array: ");
    scanf("%d", &size);

    int arr[size];

    printf("Enter the elements of the sorted array in ascending order:\n");
    for (int i = 0; i < size; i++) {
        scanf("%d", &arr[i]);
    }

    printf("Enter the a element to search for: ");
    scanf("%d", &a);

    int result = binarySearch(arr, 0, size - 1, a);

    if (result != -1) {
        printf("Element %d found at index %d\n", a, result);
    } else {
        printf("Element %d not found in the array\n", a);
    }

    return 0;
}

File: 5th Sem/ADA/2_1.py
# Implmentation and Time Analysis of Linear Search Algorithm

#Importing Time Module for Execution Time Analysis
import time

# Record start time
start = time.time()

#Take Number of elements as input from the user
n = int(input("Enter the number of elements: "))

# Take the Elements in the Array from the user
arr = []
for i in range(n):
    arr.append(int(input("Enter the element: ")))

# Linear Search Algorithm
def linearSearch(arr, n, x):
    for i in range(n):
        if arr[i] == x:
            return i
    return -1

# Array Found at Index
print("Element Found at Index: ", linearSearch(arr, n, int(input("Enter the element to search: "))))

# Record end time for Recursive method
end = time.time()

# Print the Execution Time
print("The time of execution of above program is :",(end-start) * 10**3, "ms")